backend/src/main/java/gov/cdc/usds/simplereport/api/testresult/TestResultDataResolver.java

```diff
@@ -37,18 +35,9 @@ public TestResultDataResolver(
         .withName("patientLinkDataLoader")
         .registerMappedBatchLoader(
             (testOrderIds, batchLoaderEnvironment) -> {
-              Map<UUID, List<PatientLink>> foundAll =
-                  patientLinkRepository.findAllByTestOrderInternalIdIn(testOrderIds).stream()
-                      .collect(Collectors.groupingBy(PatientLink::getTestOrderId));
-
               Map<UUID, PatientLink> found =
-                  testOrderIds.stream()
-                      .map(
-                          to ->
-                              PatientLinkDataLoader.getMostRecentPatientLink(
-                                  foundAll.getOrDefault(to, null)))
+                  patientLinkRepository.findMostRecentByTestOrderIdIn(testOrderIds).stream()
```

Comment:
After testing on demo, turns out not all test events have patient links, I remember there was a period of time when we were only creating `patientLinks` if you elect to send results via `email/sms`, I cant remember if we ever back-filled prod, but we surely didnt back-fill demo, so I am making the `patientLinkDataLoader` expect not all `testevents` will have a `patientLink` and also switched it to use SQL instead of java, so its gonna be a lot faster,